### ISSUE TITLE
Image Pull Error Workaround with Istio Gateway Deployment via Argo CD

### ISSUE THREAD

**Issue by josephaw1022** [istio/istio#53290]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

## Background

Currently I am creating a kind cluster, installing argo into the cluster, and then running a helm chart to deploy a bunch of helm charts. one of those being istio. well technically I am deploy 3 argo apps and am using sync waves to synchronize it. I know that people typically install the gateway first and then install argo so they can use the ingress, but I am not planning on using the argo ui in prod and want to keep keeping everything installed into the cluster via argo apps

This works fine and all until I got to the ingress gateway. 

![image](https://github.com/user-attachments/assets/54a1fa51-1602-463c-823d-c36b274c746a)

![image](https://github.com/user-attachments/assets/82b1040c-27c3-44ce-a3b3-794f7267294b)

Now I did see this in the readme of the ingress gateway at https://github.com/istio/istio/tree/master/manifests/charts/gateway#image-auto-information

![image](https://github.com/user-attachments/assets/eb8c29d4-dab5-4779-9767-29259a47a5a5)

however I was able to get it to work and get the image from the validation webhook by deleting the pod manually 

![image](https://github.com/user-attachments/assets/1deae579-09be-4924-ae5f-3c502f585e02)

Once I deleted it, it seemed to work 

![image](https://github.com/user-attachments/assets/7cc5b544-72f4-4f6a-a5ad-486ee7acc918)

here is a link to the repo, folder, and the correct branch

https://github.com/josephaw1022/BootstrapClusterViaGitops/tree/new-argo-format/BootstrapCluster

## Question

Now I did see this in the readme of the ingress gateway at https://github.com/istio/istio/tree/master/manifests/charts/gateway#image-auto-information

![image](https://github.com/user-attachments/assets/eb8c29d4-dab5-4779-9767-29259a47a5a5)

My question is, is there a way or workaround to this? 

Would using a pullable image mess up the injection and usage of the docker.io/istio/proxyv2:1.23.2 image in the pod? 

or could it work if you just use the docker.io/istio/proxyv2:1.23.2 image?

Two work arounds to this are to make the install of the ingress gateway imperative, so just enter the helm commands and the second is by manually deleting the pod. 

while these work for now, I want to try and keep my setup as declarative as possible. 

I was initially thinking and conflicting (sort of still am) if that this was an issue on the argo side, but to my knowledge, I haven't seen any helm charts implement something like this (I could totally be wrong), so I can't help but think there has to be some other way or workaround to avoid using image: auto. Not saying it couldnt be something that I am doing wrong on my end either. Just not sure.

If it's not possible just due to the way everything is setup, then it is what it is, but I have no idea how simple or complex this issue would be to solve.

### Version

```Text
Not using istioctl in this context. Just using helm

PS C:\Users\...> kubectl version
Client Version: v1.31.0
Kustomize Version: v5.4.2
Server Version: v1.30.0

PS C:\Users\...> helm version
version.BuildInfo{Version:"v3.15.4", GitCommit:"fa9efb07d9d8debbb4306d72af76a383895aa8c4", GitTreeState:"clean", GoVersion:"go1.22.6"}
```

**Comment by zirain** [comment]
cc @hanxiaop

**Comment by josephaw1022** [comment]
Thank you so much! Had a feeling it was something on my end. That makes sense. I'm assuming the whole point of using auto and not a specific image is so that you can upgrade istio and not have to keep up with updating the images as well and all that?

**Comment by markusobsil** [comment]
Exactly, details are [here](https://istio.io/latest/docs/setup/additional-setup/sidecar-injection/#customizing-injection) in the docs.
